src/pages/Statistic/hooks/useDateFilteredReviews.ts

```diff
@@ -2,25 +2,25 @@ import { useState, useMemo } from "react";
 import { Review } from "../../../types/review";
 import { getCurrentDateInfo } from "../../../utils/dateFormat";
 
-export const useDateFilteredReviews = (readReviews: Review[]) => {
-  // 읽은 책 중에서 연도만 담은 배열
-  const startYears = useMemo(() => readReviews.map(review => review.startDate.slice(0, 4)), [readReviews]);
+// 읽은 책 리뷰의 날짜에서 중복 값을 제거한 나머지를 추출하는 함수
+const extractUniqueValues = (reviews: Review[], sliceFn: (date: string) => string) => 
+  [...new Set(reviews.map(review => sliceFn(review.startDate)))];
+
+// 읽은 책 리뷰의 날짜에서 가장 최근 날짜를 찾는 함수
+const findMostRecent = (values: string[], defaultValue: string) => 
+  values.length ? values.reduce((max, current) => current > max ? current : max) : defaultValue;
```

Comment:
단순히 최댓값 구하기 위한 로직이라면, 아래와 같이 간단하게 작성도 가능하답니다~!

```suggestion
  values.length ? Math.max(...values) : defaultValue;
```

Reply:
그리고 여기서는 values로 들어오는 형식이 자리수가 모두 같도록 표현되는 형식으로 통일되어 있는 것 같지만, 만약 실수로 형식을 맞춰서 작성하지 않고 자리수가 달라지는 경우에는 어떻게 될까요?

```js
['5', '1', '10'].reduce((max, current) => (current > max ? current : max));
```

위의 코드의 결과 값이 어떻게 될 지, 원하는 결과 값이 나오는 지도 한 번 생각해보면 좋을 것 같아요!